doc/source/whatsnew/v2.0.0.rst

```diff
@@ -433,9 +433,11 @@ Removal of prior version deprecations/changes
 - Remove :meth:`DataFrameGroupBy.pad` and :meth:`DataFrameGroupBy.backfill` (:issue:`45076`)
 - Remove ``numpy`` argument from :func:`read_json` (:issue:`30636`)
 - Disallow passing abbreviations for ``orient`` in :meth:`DataFrame.to_dict` (:issue:`32516`)
+- Disallow partial slicing on an unordered :class:`DatetimeIndex` with keys, which are not in Index. This now raises a ``KeyError`` (:issue:`18531`)
```

Comment:
"keys, which are not in Index" -> "keys which are not in the Index"?
"unordered" -> "non-monotonic"?

Reply:
Thanks. Updated + greenish